Automattic/jetpack
# Jetpack connection usage update

#### Changes proposed in this Pull Request:

This PR updates the usage of the connection package using the proxy method instead of calling individual methods for each step of the disconnection routine.

The new methods make sure we fire the proper hooks and trigger stats and tracks events.

There should be no functional changes.

#### Jetpack product discussion


p9dueE-4n9-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Check connection and disconnection work as expected

Confirm reconnection still works
* Connect Jetpack and activate a secondary plugin like Boost
* Go to Jetpack Debug Tools > Broken Token and "Set an invalid blog token"
* Go to Jetpack dashboard
* You should see an error message
* Click "Restore connection"
* Make sure the connection is restored
* We just tested that, in this case, Jetpack is hard disconnecting the site even though there's another plugin using the connection

Confirm CLI still works
* Connect Jetpack and activate a secondary plugin like Boost
* Run the CLI command `jetpack docker wp jetpack disconnect blog`
* Confirm the site was disconnected
* We just tested that, in this case, Jetpack is hard disconnecting the site even though there's another plugin using the connection

Confirm XMLRPC endpoint still works
* Connect Jetpack and activate a secondary plugin like Boost
* Go to the Online Jetpack debugger for your site
* on the "Blog Details" section, click on "Disconnect" to disconnect your site
* Go back to your site and make sure it got disconnected
* We just tested that, in this case, Jetpack is hard disconnecting the site even though there's another plugin using the connection
